### ISSUE TITLE
System.Linq.Dynamic.Core.Exceptions.ParseException : No applicable method 'DateTime' exists in type

### ISSUE THREAD

**Issue by ascott18** [zzzprojects/System.Linq.Dynamic.Core#662]
## 1. Description
I am unable to use the documented constructor syntax when the type being constructed has the same name as a property on the queried type. See fiddle. For example - `.Where("it.DateTime > DateTime(2022, 1, 1, 0, 0, 0)")`

This worked in versions at or prior to 1.0.22. It was broken in 1.0.23, presumably by https://github.com/zzzprojects/System.Linq.Dynamic.Core/pull/357. It continues to be broken in 1.2.24.

## 2. Exception

```
  Message: 
System.Linq.Dynamic.Core.Exceptions.ParseException : No applicable method 'DateTime' exists in type 'Foo'

  Stack Trace: 
ExpressionParser.ParseMemberAccess(Type type, Expression expression)
ExpressionParser.ParseIdentifier()
ExpressionParser.ParsePrimaryStart()
ExpressionParser.ParsePrimary()
ExpressionParser.ParseUnary()
ExpressionParser.ParseMultiplicative()
ExpressionParser.ParseAdditive()
ExpressionParser.ParseShiftOperator()
ExpressionParser.ParseComparisonOperator()
ExpressionParser.ParseLogicalAndOrOperator()
<20 more frames...>
ExpressionParser.ParseConditionalOperator()
ExpressionParser.Parse(Type resultType, Boolean createParameterCtor)
DynamicExpressionParser.ParseLambda(Type delegateType, ParsingConfig parsingConfig, Boolean createParameterCtor, ParameterExpression[] parameters, Type resultType, String expression, Object[] values)
DynamicExpressionParser.ParseLambda(ParsingConfig parsingConfig, Boolean createParameterCtor, ParameterExpression[] parameters, Type resultType, String expression, Object[] values)
DynamicExpressionParser.ParseLambda(ParsingConfig parsingConfig, Boolean createParameterCtor, Type itType, Type resultType, String expression, Object[] values)
DynamicQueryableExtensions.Where(IQueryable source, ParsingConfig config, String predicate, Object[] args)
DynamicQueryableExtensions.Where[TSource](IQueryable`1 source, ParsingConfig config, String predicate, Object[] args)
DynamicQueryableExtensions.Where[TSource](IQueryable`1 source, String predicate, Object[] args)
```

## 3. Fiddle or Project

https://dotnetfiddle.net/6j1G9U

**Comment by StefH** [comment]
Just a quick remark:

Did you try escaping the property DateTime like:


Where("@DateTime

**Comment by ascott18** [comment]
The exception is being throw about the constructor invocation `DateTime(2022, 1, 1, 0, 0, 0)`, not the property access `it.DateTime`.

**Comment by StefH** [comment]
@ascott18 
The only way to solve this is to introduce a new config setting, which will have value false = old behavior. And set this to true when you want the new behavior (priority property over the type)

https://github.com/zzzprojects/System.Linq.Dynamic.Core/pull/664

A new NuGet version will be released in some time.

**Comment by StefH** [comment]
@ascott18 
This new config setting causes some backward compatibility issues some some users, so I'll probably need to change the default behavior back to version 1.2.24

So when a new version is released, 1.3.x, you 'll need to use the config setting correctly.
